modules/createHistory.js

```diff
@@ -114,6 +114,16 @@ function createHistory(options={}) {
         return // Transition was interrupted.
 
       if (ok) {
+        // treat PUSH to current path like REPLACE to be consistent with browsers
+        if (nextLocation.action === PUSH) {
+          let { pathname, search } = getCurrentLocation()
```

Comment:
I'm not 100% sure this is the right place to do this, but should be ok for now. My concern is for environments where we may not be able to synchronously call `getCurrentLocation`, like native, where we have to use `AsyncStorage` to store everything.